model/Service/ConcurringSessionService.php

```diff
@@ -298,20 +308,27 @@ private function getContextByDeliveryExecution(DeliveryExecutionInterface $execu
         );
     }
 
-    private function getTimerAdjustmentService(): TimerAdjustmentService
+    private function getTimerAdjustmentService(): TimerAdjustmentServiceInterface
     {
-        /** @noinspection PhpIncompatibleReturnTypeInspection */
-        return $this->getServiceManager()->get(TimerAdjustmentService::SERVICE_ID);
+        if (!$this->timerAdjustmentService instanceof TimerAdjustmentServiceInterface) {
+            /** @noinspection PhpFieldAssignmentTypeMismatchInspection */
+            $this->timerAdjustmentService = ServiceManager::getServiceManager()->get(
+                TimerAdjustmentServiceInterface::SERVICE_ID
+            );
+        }
+
+        return $this->timerAdjustmentService;
     }
 
     private function getTestSessionService(): TestSessionService
     {
-        /** @noinspection PhpIncompatibleReturnTypeInspection */
-        return $this->getServiceManager()->get(TestSessionService::SERVICE_ID);
-    }
+        if (!$this->testSessionService instanceof TestSessionService) {
```

Comment:
Why cant we rely on DI container? I know that `$this->testSessionService` is nullable, but this is decided in the DI Container definition files, not here. I would remove completely the `ServiceManager` usage from this class. Wdyt?

Reply:
I know we did it here, but maybe we can avoid that hybrid approach? 

https://github.com/oat-sa/extension-tao-testqti/commit/d66b9e501cedf0fd7978278e9787ae7e250202ed#diff-be1605fd62af343d118c742ba9c27603c98978aeff03b929a54110c9ba97cda6R256-R266

Reply:
Makes sense, changed here: 024d5dd01

Reply:
Thanks!